veevavault/1.4/modules/ROOT/pages/veevavault-connector-studio.adoc

```diff
@@ -0,0 +1,208 @@
+----
+|API Limit Threshold | Veeva Vault API limit threshold for *Daily API Limit Profile*. When the specified limit is reached or exceeded, the connector waits for the specified amount of time. The default is `10000`.
+|===
+
+The following image shows an example of configuring the *Advanced* tab for Basic authentication:
+
+image::basic-auth-adv.png[Authentication fields completed in the *Advanced* tab for Basic authentication]
+
+
+[[oauth2-oidc-connection]]
+=== OAuth 2.0 / Open ID Connection Provider
+
+Enter the following information in the *General* tab of the *Global Element Properties* screen to configure OAuth 2.0 / Open ID connection:
+
+[%header,cols="30s,70a"]
+|===
+|Field |Description
+|Name | Configuration name
+|Vault URL | Veeva Vault instance URL required for connection.
+|Version | Version for the Veeva Vault instance required.
+|Client Id | Client ID required to send with each request header to the Vault instance.
+|OIDC Profile ID | OIDC profile ID generated by Veeva when creating a new OIDC profile. To configure the OIDC profile ID, use the ID in your OIDC profile in Veeva. Your OIDC profile contains a *Vault Session ID URL*, for example, `\https://login.veevavault.com/auth/oauth/session/_00000000-aaaa-bbbb-cccc-111111111111` where `_00000000-aaaa-bbbb-cccc-111111111111`. Use this value for the *OIDC Profile ID* field.
+|Consumer Key | OAuth consumer key, as registered with the service provider. This is your IdP client ID, assuming you already have a client application registration created at the ID provider of your choice, for example, PoneOne or PingFederate.
+|Consumer Secret | OAuth consumer secret, as registered with the service provider. This is your IdP client secret, assuming you already have a client application registration created at the ID provider of your choice, for example, PoneOne or PingFederate.
+|Authorization Url | OAuth2 authorization URL, such as `\https://{your-instance}/as/authorization.oauth2` for PingFederate.
+|Access Token Url | OAuth2 token URL, such as `\https://{your-instance}/as/token.oauth2`.
+|Scopes | You can set up your OAuth2 scopes as necessary, but the important ones `openid` and `offline_access` are already pre-filled.
+|Resource Owner Id | Depending on your IdP configuration, you can leave it blank.
+|Listener Config | Create a new HTTP Listener or use an existing one if you have one, for example, `\http://localhost:8081/`.
+|Callback Path | Callback path of your application. If you set it to `callback`, your full callback path is `\http://localhost:8081/callback`. Do not forget to add your callback path to the IdP's redirect URLs.
+|Authorize Path | Authorize path. You will navigate to this path when starting the OAuth 2.0 dance, for example, `authorize` means the full path is `\http://localhost:8081/authorize`.
+|External Callback Url | You can set this directly to `\http://localhost:8081/callback`.
+|===
+
```

Comment:
I'm not sure where exactly, but we should add a more thorough guide on how to configure and authorize an OAuth flow - what is missing here is that the user has to also configure an HTTP listener, mentioning how the Mule uses the listener for doing the OAuth dance and how the user needs to authorize first before making any requests.
maybe we should add this guide post-release and then reuse it across multiple OAuth connectors?
CC @Hazarth

Reply:
I agree. Really we should have an entire page somewhere in the Mule Docs just about OAuth and setting up OAuth for CloudHub and connectors should just Link to that page using a _Learn More_ link

OAuth setup is an entire big thing and it keeps evolving, for example we don't even support all the OAuth type configurations which should definitely be documented somewhere. The docs should probably also be written by someone with plenty of experience with OAuth and only then reviewed and finished up by a technical writer 🤔

Could we get an entire page dedicated just for OAuth somewhere @sathya0 ? Or does such a page already exists? If yes then it should be linked here, but if not, we should allocate time to create one, because we found out that this is a very confusing issue for customers that don't have experience with OAuth

CC @adam-lipka

Reply:
hey @Hazarth and @adam-lipka, happy to add this page into our docs. I agree we can link to this page from the Authorize Path parameter. This info might be documented somewhere in our Mule docs. Do these pages cover this info? 

https://docs.mulesoft.com/mule-sdk/1.1/oauth-configuring 
https://docs.mulesoft.com/mule-sdk/1.1/oauth-dance
https://docs.mulesoft.com/mule-sdk/1.1/authorization-code

Reply:
@sathya0 This one https://docs.mulesoft.com/mule-sdk/1.1/oauth-dance covers it, but there are a few issues IMO: The first is that the doc page is hidden under the Mule SDK section and which is targeted for connector developers, not for connector users, so they have little chance of finding that page. 
The second problem is that it seems that some customers need to have more explicit explanation of what to do - probably even something like:
- hit the xxxx URL in a browser
- login in a login window
- now the connector is authenticated

Then it lacks the coverage of ClougHub x OAuth configuration, which comprises multiple things you have to set up and would be nice to have in under one page.